src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs

```diff
@@ -203,6 +203,38 @@ private static bool CanBindToken(SyntaxToken token)
         public static string GenerateNameForExpression(
             this SemanticModel semanticModel, ExpressionSyntax expression, bool capitalize = false)
         {
+
+            // Check if expression is ArgumentSyntax. If we have an expression that
+            // is an argument, return the name of the parameter.
+            var parent = expression.Parent as ArgumentSyntax;
+            if (parent != null)
+            {
+                if (parent.NameColon != null)
+                {
+                    return parent.NameColon.Name.Identifier.ValueText.ToCamelCase();
```

Comment:
1. Do we care if the name is missing?
 2. Do we care if the name is wrong?
 3. Why do we need to call ToCamelCase?  Shouldn't we follow the casing of the named argument?